gulpfile.js

```diff
@@ -153,31 +179,18 @@ gulp.task('test-react-warnings', function() {
     gutil.log(gutil.colors.red.bold(message));
   };
 
-  return new IndexFileStream(require('./lib/index-static.jsx'))
+  return createIndexFileStream()
     .on('end', function() {
       console.warn = oldWarn;
       if (warnings) {
         this.emit('error', new Error('At least one warning was logged.'));
       }
     })
-    .pipe(gulp.dest('./dist'));
+    .on('data', function() {});
```

Comment:
Ah yeah, that is because `pipe`'ing a readable stream into another stream puts it into flowing mode (which is what's happening in the [example](https://github.com/mozilla/teach.webmaker.org/pull/839/files#diff-b9e12334e9eafd8341a6107dd98510c9R229) you linked to). It additionally allows node to automatically regulate how quickly it reads from the stream through **back pressure**, which is nicely explained in Tim Caswell's [Daddy, what's a stream?](http://howtonode.org/streams-explained) post.

Basically with a stream, you want to either add a `data` event listener *or* pipe it somewhere--otherwise node won't bother reading from it, because it doesn't think anyone's listening to it, and it also doesn't want to exhaust  available memory by buffering it all up. It's not very common to have a `data` event listener that does nothing, but we want to do it here because we're really just trying to see if any errors were logged while the readable stream was doing its thing--we don't care about the actual *output* of the stream, which is a bit unusual.